File: src/main/ruby/scpTpkg.rb
#=====
# Main
#=====
if ARGV.length < 1 then
  print "Usage: ruby scpTpkg.rb remoteHost\n"
  Process.exit(1)
end

remoteHost = ARGV[0]
tpkgFile = "tpkgFile"
remoteDestination = "@#{remoteHost}:centos-tpkg/."
puts "remoteDestination=#{remoteDestination}"

# find all the tpkg files
cmd = "find . -name \"*.tpkg\" -print > " << tpkgFile
puts cmd
system(cmd)

# read and copy them to remote server
open(tpkgFile).each { |line|
  line.strip!
  cmd = "scp #{line} #{remoteDestination}"
  puts cmd
  system(cmd)
}

File: src/main/python/scpTpkg.py
import os
import sys

#=====
# Main
#=====
if (len(sys.argv) != 2):
    print "Usage: python scpTpkg.py remoteHost"
    sys.exit(-1)

remoteHost = sys.argv[1]
tpkgFile = "tpkgFile"
remoteDestination = "@%s:centos-tpkg/." % (remoteHost)
print "remoteDestination=%s" % (remoteDestination)

# find all the tpkg files
cmd = "find . -name \"*.tpkg\" -print > %s" % (tpkgFile)
print cmd
os.system(cmd)

# read and copy them to remote server
infile = open(tpkgFile,"r")
line = infile.readline()
while line:
    if line[-1] == "\n":
        line = line[:-1]
    cmd = "scp %s %s" % (line, remoteDestination)
    print cmd
    os.system(cmd)
    line = infile.readline()

infile.close()
